### PR TITLE
Add support for vector-valued objective functions

### PR DESCRIPTION
**Pull request by odow**
x-ref: https://github.com/jump-dev/MathOptInterface.jl/pull/2070
x-ref: https://github.com/jump-dev/JuMP.jl/issues/2099
x-ref: https://github.com/jump-dev/JuMP.jl/pull/3176

### CODE REVIEW

File: test/MOI/MOI_multiobjective.jl [hunk 1]
```diff
@@ -55,7 +55,7 @@ c4: y >= 0.25
     @test MOI.get(model, Gurobi.MultiObjectivePriority(2)) == 0
 
     MOI.optimize!(model)
-    @test MOI.get(model, MOI.ObjectiveValue()) ≈ 1.5
+    @test MOI.get(model, MOI.ObjectiveValue()) ≈ [1.5, 2.0]
```
Comment (odow) [inline test/MOI/MOI_multiobjective.jl:58]: **This is a breaking change** (albeit, a minor one for someone using the multi-objective interface to Gurobi)

Comment (odow) [reply test/MOI/MOI_multiobjective.jl:58]: @blegat: this is a good enough time as any to release a Gurobi v1.0.0?

Comment (blegat) [reply test/MOI/MOI_multiobjective.jl:58]: I don't think we plan any more breaking changes so I don't see any objection

Comment (odow) [reply test/MOI/MOI_multiobjective.jl:58]: Okay. I'll merge and then make a few other changes before tagging Gurobi v1.0

**Pull request merged**